buildkite/resource_organization_rule.go

```diff
@@ -0,0 +1,370 @@
+		)
+		return
+	}
+
+	// Convert fron Node to getNodeNodeRule type
+	if organizationRule, ok := apiResponse.GetNode().(*getNodeNodeRule); ok {
+		if organizationRule == nil {
+			resp.Diagnostics.AddError(
+				"Unable to get organization rule",
+				"Error getting organization rule: nil response",
+			)
+			return
+		}
+		// Update organization rule model and set in state
+		updateOrganizatonRuleReadState(&state, *organizationRule)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	} else {
+		// Remove from state if not found{{}}
+		resp.Diagnostics.AddWarning("Organization rule not found", "Removing from state")
+		resp.State.RemoveResource(ctx)
+		return
+	}
+}
+
+func (or *organizationRuleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (or *organizationRuleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError("Cannot update an organization rule", "An existing rule must be deleted/re-created")
+	panic("cannot update an organization rule")
+}
+
+func (or *organizationRuleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state organizationRuleResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := or.client.timeouts.Delete(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := retry.RetryContext(ctx, timeout, func() *retry.RetryError {
+		org, err := or.client.GetOrganizationID()
+		if err == nil {
+			_, err = deleteOrganizationRule(
+				ctx,
+				or.client.genqlient,
+				*org,
+				state.ID.ValueString(),
+			)
+		}
+
+		return retryContextError(err)
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to delete organization rule",
+			fmt.Sprintf("Unable to delete organization rule: %s", err.Error()),
+		)
+		return
+	}
+}
+
+func obtainCreationUUIDs(r *createOrganizationRuleResponse) (*string, *string, error) {
+	var sourceUUID, targetUUID string
+	/*
+		The provider will try and determine the source UUID based on type that is returned in the *createOrganizationRuleResponse
+		It will switch based on the SourceType returned in the response and extract the UUID of the respective source based on this.
+		Otherwise, it will create and throw an error stating that it cannot obtain the source type from the returned API response.
+	*/
+	switch r.RuleCreate.Rule.SourceType {
+	case "PIPELINE":
+		if ruleCreateSourcePipeline, ok := r.RuleCreate.Rule.Source.(*OrganizationRuleFieldsSourcePipeline); ok {
+			sourceUUID = ruleCreateSourcePipeline.Uuid
+		} else {
+			return nil, nil, errors.New("Error obtaining source type upon creating the organization rule.")
+		}
+	}
```

Comment:
Without a default block, the code will continue to the next section. This means that if the case is NOT `PIPELINE` it will continue to the next block, which also checks if it's `PIPELINE`, which will then continue on and try to return a `&sourceUUID` and `&targetUUID` which I _think_ will cause a nil pointer error.

Reply:
I've added a `default` to both thees switch statements to catch when GraphQL doesn't return the type for TF to store in state, and thus will error 👍